packages/kbn-build/src/utils/Package.js

```diff
@@ -0,0 +1,125 @@
+import path from 'path';
+import chalk from 'chalk';
+
+import {
+  runScriptInDir,
+  runScriptInPackageStreaming,
+  installInDir
+} from './npm';
+import { readPackageJson } from './packageJson';
+import { CliError } from './errors';
+
+export class Package {
+  static async fromPath(path) {
+    const pkgJson = await readPackageJson(path);
+    return new Package(pkgJson, path);
+  }
+
+  constructor(packageJson, packageDir) {
+    this._json = packageJson;
+    this.path = packageDir;
+
+    this.packageJsonLocation = path.resolve(this.path, 'package.json');
+    this.nodeModulesLocation = path.resolve(this.path, 'node_modules');
+    this.targetLocation = path.resolve(this.path, 'target');
+  }
+
+  get name() {
+    return this._json.name;
+  }
+
+  get version() {
+    return this._json.version;
+  }
+
+  get dependencies() {
+    return this._json.dependencies || {};
+  }
+
+  get devDependencies() {
+    return this._json.devDependencies || {};
+  }
+
+  get allDependencies() {
+    return {
+      ...this.devDependencies,
+      ...this.dependencies
+    };
+  }
+
+  get scripts() {
+    return this._json.scripts || {};
+  }
+
+  ensureValidPackageVersion(pkg, version) {
+    const relativePathToPkg = path.relative(this.path, pkg.path);
+    const expectedVersion = `link:${relativePathToPkg}`;
+
+    const expectedValue = `"${pkg.name}": "${expectedVersion}"`;
+    const actualValue = `"${pkg.name}": "${version}"`;
```

Comment:
__nit:__ maybe do an early return here? It will simplify `if/else` for `CliError`s below as well.
```js
if (version.startsWith("link:") && version === expectedVersion) {
  return;
}

const meta = { ... };
const updateMsg = "Update it's package.json to the expected value below.";

if (version.startsWith("link:")) {
  throw new CliError.....
}

throw new CliError....
```